src/common/hooks/useEndorsementChain/fetchEscrowTransfer.ts

```diff
@@ -54,3 +54,19 @@ export const fetchEventInfo = async (
     eventTimestamp,
   };
 };
+
+export const fetchEvents = async (
+  address: string,
+  blockNumber: number,
+  provider: providers.Provider
+): Promise<TradeTrustErc721Event> => {
+  try {
+    return await fetchEscrowTransfers(address, provider);
+  } catch (e) {
+    //SupportsInterface throws a call revert exception when the owner is a wallet
+    if (e.message.includes("call revert exception")) {
```

Comment:
This is actually a very generic error. The reason this error happens is that it tries to query that address as if its a smart contract but when it calls `supportsInterface` it fails. This means that this function has a hidden coupling with the order of the calls in the above function. An example of how this flow could fail:

The address is a smart contract that has the supported interface, but for some reason maybe the `beneficiary` method throws call revert because it is not implemented. Now this code will assume that the contract is an EOA.

A better way to do this is to check if the address has code, u can do this via `provider.getCode` i think. If there is no code, you know for sure it is not a smart contract and may assume that its "Transferred to wallet". If there is code, then just proceed to `fetchEscrowTransfers`